analyzere/base_resources.py

```diff
@@ -148,6 +149,13 @@ def to_dict(self):
             d[k] = to_dict(v)
         return d
 
+    def __hash__(self):
+        # Certainly not the most efficient hash, since anything without an ID
```

Comment:
There was no default hash implementation (AFAIK). Outside libraries were complaining that the object was "unhashable".